packages/terra-base/src/Base.jsx

```diff
@@ -1,20 +1,92 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { I18nProvider, i18nLoader } from 'terra-i18n';
 import './baseStyles';
 
 const propTypes = {
   /**
    * The component(s) that will be wrapped by `<Base />`.
    */
   children: PropTypes.node.isRequired,
+  /**
+   * The locale name.
+   */
+  locale: PropTypes.string,
+  /**
+   * Customized translations provided by consuming application
+   * only for current locale.
+   */
+   /* eslint-disable consistent-return */
+  customMessages: (props, propName, componentName) => {
+    if (Object.keys(props[propName]).length !== 0 && props.locale === undefined) {
+      return new Error(
+        `Missing locale prop for ${propName} in ${componentName} props`,
+      );
+    }
+  },
+  /**
+   * Component used as PlaceHolder when the translations loading.
+   */
+  loadingPlaceholder: PropTypes.node,
+};
+
+const defaultProps = {
+  customMessages: {},
+  loadingPlaceholder: null,
 };
 
-const Base = ({ children, ...customProps }) => (
-  <div {...customProps}>
-    {children}
-  </div>
-);
+class Base extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      areTranslationsLoaded: false,
+      locale: props.locale,
+      messages: {},
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.locale !== undefined) {
+      i18nLoader(this.props.locale, this.setState, this);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props === nextProps) return;
+    if (nextProps.locale !== undefined) {
+      i18nLoader(nextProps.locale, this.setState, this);
+    }
+  }
+
+  render() {
+    const {
+      children,
+      locale,
+      customMessages,
+      loadingPlaceholder,
+      ...customProps
+    } = this.props;
+
+    const childComponent = (
+      <div {...customProps}>
+        {children}
+      </div>
+    );
+
+    const messages = Object.assign({}, this.state.messages, customMessages);
+
+    if (locale === undefined) return childComponent;
+    if (!this.state.areTranslationsLoaded) return loadingPlaceholder;
```

Comment:
As we currently don't have a loading spinner to indicate the loading, I can only test by faked component(like label). After a discussion with Brett, I will remove the `loadingPlaceholder` for  now. As it's a quite easy prop, I will add it back when we have the loading indicator, or some consumers really require this feature.